File: day1.cpp
# include <iostream>
# include <vector>
# include <string>

using namespace std;

int main(){
    string lineInp;
    long long total_sum = 0;
    int count = 0;

    vector<string> numbers;
    numbers.push_back("one");
    numbers.push_back("two");
    numbers.push_back("three");
    numbers.push_back("four");
    numbers.push_back("five");
    numbers.push_back("six");
    numbers.push_back("seven");
    numbers.push_back("eight");
    numbers.push_back("nine");


    while(getline(cin, lineInp)){
        if(lineInp.empty()) {
            break;
        }

        long long first_digit = -1, last_digit = -1;
        for(int i=0; i < lineInp.length(); i++){
            char ch = lineInp[i];

            if(ch >= '0' && ch <= '9'){
                last_digit = ch-'0';

                if(first_digit == -1)
                    first_digit = ch-'0';
            } else {

                for(int number_idx=0; number_idx < numbers.size(); number_idx++){

                    if(lineInp.substr(i, numbers[number_idx].length()) == numbers[number_idx]) {
                        

                        last_digit = number_idx+1;

                        if(first_digit == -1)
                            first_digit = number_idx+1;
                    }
                }

            }
        }

        if(first_digit == -1)
            cout << "No digit in string\n";
        total_sum += first_digit * 10 + last_digit;
        cout << "current 2 digit number = " << first_digit << last_digit << endl;
        cout << "current total sum = " << total_sum << endl;

    }
    cout << "total sum = " << total_sum << endl;

}

File: advent-of-code-2023/aoc_py/day1.py
with open('input') as f:
    lines = f.readlines()

total = 0
for line in lines:
    num = ""
    for ch in line:
        if ch.isnumeric():
            if num == '':
                num += ch * 2
            else:
                num = num[:1]+ch
        
    total += int(num)

print("part1 : ", total)

total = 0
numbers_spelled_out = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']

for line in lines:
    num = ""
    for idx in range(len(line)):
        
        strs = []
        ch = line[idx]
        for s in [3, 4, 5]:
            if line[idx:min(idx+s, len(line))] in numbers_spelled_out:
                ch = f'{numbers_spelled_out.index(line[idx:min(idx+s, len(line))])+1}'
                break
        
        if ch.isnumeric():
            if num == '':
                num += ch * 2
            else:
                num = num[:1]+ch
        

    total += int(num)

print("part2 : ", total)